### PR TITLE
upgrade to oras.land/oras.go/v2

### PR DESCRIPTION
**Pull request by deitch**
Replaces `oras.land/oras.go` (v1) with `oras.land/oras.go/v2`. As a result of this and the changes in it:

* No longer tries to get a `remotes.Resolver` from `oras.land/oras-go/pkg/auth/docker` but rather from `github.com/containerd/containerd/remotes/docker.NewResolver`, which was the fallback method before. oras v2 does not return a `containerd/remotes.Resolver` anymore; it is completely clean of anything `containerd`
* Updates many other dependencies to more recent versions, especially `docker/docker`, and simplifies `go.mod`
* Changed reference to the oras memory store to be compliant with v2

While at it, cleaned up a lint dependency and commented some others, so it will pass linting.

Also updated `yaml.v2` to `yaml.v3`, which, in turn, needed a more recent version of go.

### CODE REVIEW

**Comment by deitch** [comment]
Note: I did not run any real functional tests, not sure if the CI here does it.

**Comment by deitch** [comment]
Derivative note: both of the big libraries that interact directly with registries - oras and ggcr - now avoid containerd library entirely. This is primarily due to the sheer size of the import stack, as far as I know. Either way, perhaps manifest-tool should use one of them more directly and simply push/inspect/pull manifests directly rather than using the `containerd/remotes.Resolver` and the complex handler chain?

**Comment by deitch** [comment]
I was in error; that fix did not get lint to pass. I even tried bumping to latest golangci-lint v1.50.1, no help. The oddest error [here](https://github.com/estesp/manifest-tool/actions/runs/4023046289/jobs/6913457873#step:4:23):

```
level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package syncutil: could not load export data: no export data for \"oras.land/oras-go/v2/internal/syncutil\""
[23](https://github.com/estesp/manifest-tool/actions/runs/4023046289/jobs/6913457873#step:4:25)
  level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package syncutil: could not load export data: no export data for \"oras.land/oras-go/v2/internal/syncutil\"\n\n"
[24](https://github.com/estesp/manifest-tool/actions/runs/4023046289/jobs/6913457873#step:4:26)
```

I can run it locally, the only thing syncutil package depends upon is `sync` and `golang.org/x/sync` (and sub packages). 🤷‍♂️

**Comment by deitch** [comment]
Aha! That did it! Installed go explicitly.

**Comment by estesp** [comment]
Sorry, forgot to come back to your work here and just did some vendoring updates--created some conflicts that should be easy to resolve with a repush.

**Comment by deitch** [comment]
Sure @estesp ; done

File: v2/go.mod [hunk 1]
```diff
@@ -1,59 +1,35 @@
...
 	github.com/fatih/color v1.10.0
 	github.com/opencontainers/go-digest v1.0.0
 	github.com/opencontainers/image-spec v1.1.0-rc2
 	github.com/pkg/errors v0.9.1
-	github.com/sirupsen/logrus v1.9.0
-	github.com/urfave/cli v1.22.12
-	gopkg.in/yaml.v2 v2.4.0
-	oras.land/oras-go v1.2.0
+	github.com/sirupsen/logrus v1.8.1
+	github.com/urfave/cli v1.22.9
```
Comment (estesp) [inline v2/go.mod:14]: should be able to keep the upgrade of logrus and urfave/cli with your PR; everything else looks fine

Comment (deitch) [reply v2/go.mod:14]: Ah, did I mess it up? That often happens when dealing with rebasing go packages. I will fix it right away. Hang on.

Comment (deitch) [reply v2/go.mod]: OK, think I got it. Take another look?

Comment (estesp) [reply v2/go.mod]: yup, thanks!

**Review by estesp — APPROVED** [review]
LGTM

**Pull request merged**